#+	file, err := ioutil.ReadFile(fmt.Sprintf(h.options.LocalAssets+"/templates/%s.html", name))
--- a/web/web.go
+++ b/web/web.go
@@ -14,7 +14,6 @@
 package web
 
 import (
-	"bytes"
 	"encoding/json"
 	"fmt"
 	"io"
@@ -46,7 +45,6 @@
 	"github.com/prometheus/prometheus/template"
 	"github.com/prometheus/prometheus/util/httputil"
 	api_v1 "github.com/prometheus/prometheus/web/api/v1"
-	"github.com/prometheus/prometheus/web/ui"
 )
 
 var localhostRepresentations = []string{"127.0.0.1", "localhost"}
@@ -111,7 +109,7 @@
 	ExternalURL          *url.URL
 	RoutePrefix          string
 	MetricsPath          string
-	UseLocalAssets       bool
+	LocalAssets          string
 	UserAssetsPath       string
 	ConsoleTemplatesPath string
 	ConsoleLibrariesPath string
@@ -180,7 +178,7 @@
 
 	router.Get("/consoles/*filepath", instrf("consoles", h.consoles))
 
-	router.Get("/static/*filepath", instrf("static", serveStaticAsset))
+	router.Get("/static/*filepath", instrf("static", route.FileServe(filepath.Join(o.LocalAssets, "/static"))))
 
 	if o.UserAssetsPath != "" {
 		router.Get("/user/*filepath", instrf("user", route.FileServe(o.UserAssetsPath)))
@@ -198,28 +196,6 @@
 	return h
 }
 
-func serveStaticAsset(w http.ResponseWriter, req *http.Request) {
-	fp := route.Param(route.Context(req), "filepath")
-	fp = filepath.Join("web/ui/static", fp)
-
-	info, err := ui.AssetInfo(fp)
-	if err != nil {
-		log.With("file", fp).Warn("Could not get file info: ", err)
-		w.WriteHeader(http.StatusNotFound)
-		return
-	}
-	file, err := ui.Asset(fp)
-	if err != nil {
-		if err != io.EOF {
-			log.With("file", fp).Warn("Could not get file: ", err)
-		}
-		w.WriteHeader(http.StatusNotFound)
-		return
-	}
-
-	http.ServeContent(w, req, info.Name(), info.ModTime(), bytes.NewReader(file))
-}
-
 // ListenError returns the receive-only channel that signals errors while starting the web server.
 func (h *Handler) ListenError() <-chan error {
 	return h.listenErrCh
@@ -454,11 +430,11 @@
 }
 
 func (h *Handler) getTemplate(name string) (string, error) {
-	baseTmpl, err := ui.Asset("web/ui/templates/_base.html")
+	baseTmpl, err := ioutil.ReadFile(filepath.Join(h.options.LocalAssets, "templates/_base.html"))
 	if err != nil {
 		return "", fmt.Errorf("error reading base template: %s", err)
 	}
-	pageTmpl, err := ui.Asset(filepath.Join("web/ui/templates", name))
+	pageTmpl, err := ioutil.ReadFile(filepath.Join(h.options.LocalAssets, "templates", name))
 	if err != nil {
 		return "", fmt.Errorf("error reading page template %s: %s", name, err)
 	}
--- a/console_libraries/prom.lib
+++ b/console_libraries/prom.lib
@@ -1,14 +1,13 @@
 {{/* vim: set ft=html: */}}
 {{/* Load Prometheus console library JS/CSS. Should go in <head> */}}
 {{ define "prom_console_head" }}
-<link type="text/css" rel="stylesheet" href="{{ pathPrefix }}/static/vendor/rickshaw/rickshaw.min.css">
-<link type="text/css" rel="stylesheet" href="{{ pathPrefix }}/static/vendor/bootstrap-3.3.1/css/bootstrap.min.css">
+<link type="text/css" rel="stylesheet" href="{{ pathPrefix }}/static/rickshaw/rickshaw.min.css">
+<link type="text/css" rel="stylesheet" href="{{ pathPrefix }}/static/bootstrap/css/bootstrap.min.css">
 <link type="text/css" rel="stylesheet" href="{{ pathPrefix }}/static/css/prom_console.css">
-<script src="{{ pathPrefix }}/static/vendor/rickshaw/vendor/d3.v3.js"></script>
-<script src="{{ pathPrefix }}/static/vendor/rickshaw/vendor/d3.layout.min.js"></script>
-<script src="{{ pathPrefix }}/static/vendor/rickshaw/rickshaw.min.js"></script>
-<script src="{{ pathPrefix }}/static/vendor/js/jquery.min.js"></script>
-<script src="{{ pathPrefix }}/static/vendor/bootstrap-3.3.1/js/bootstrap.min.js"></script>
+<script src="{{ pathPrefix }}/static/d3/d3.js"></script>
+<script src="{{ pathPrefix }}/static/rickshaw/rickshaw.min.js"></script>
+<script src="{{ pathPrefix }}/static/jquery/jquery.min.js"></script>
+<script src="{{ pathPrefix }}/static/bootstrap/js/bootstrap.min.js"></script>
 
 <script>var PATH_PREFIX = "{{ pathPrefix }}";</script>
 <script src="{{ pathPrefix }}/static/js/prom_console.js"></script>
--- a/web/ui/templates/graph.html
+++ b/web/ui/templates/graph.html
@@ -1,18 +1,18 @@
 {{define "head"}}
     <link type="text/css" rel="stylesheet" href="{{ pathPrefix }}/static/css/graph.css">
 
-    <link type="text/css" rel="stylesheet" href="{{ pathPrefix }}/static/vendor/rickshaw/rickshaw.min.css">
-    <link type="text/css" rel="stylesheet" href="{{ pathPrefix }}/static/vendor/bootstrap-datetimepicker/bootstrap-datetimepicker.min.css">
+    <link type="text/css" rel="stylesheet" href="{{ pathPrefix }}/static/rickshaw/rickshaw.min.css">
+    <link type="text/css" rel="stylesheet" href="{{ pathPrefix }}/static/eonasdan-bootstrap-datetimepicker/bootstrap-datetimepicker.min.css">
 
-    <script src="{{ pathPrefix }}/static/vendor/rickshaw/vendor/d3.v3.js"></script>
-    <script src="{{ pathPrefix }}/static/vendor/rickshaw/vendor/d3.layout.min.js"></script>
-    <script src="{{ pathPrefix }}/static/vendor/rickshaw/rickshaw.min.js"></script>
-    <script src="{{ pathPrefix }}/static/vendor/bootstrap-datetimepicker/bootstrap-datetimepicker.js"></script>
-    <script src="{{ pathPrefix }}/static/vendor/bootstrap3-typeahead/bootstrap3-typeahead.min.js"></script>
+    <script src="{{ pathPrefix }}/static/d3/d3.min.js"></script>
+    <script src="{{ pathPrefix }}/static/rickshaw/rickshaw.min.js"></script>
+    <script src="{{ pathPrefix }}/static/moment/moment.min.js"></script>
+    <script src="{{ pathPrefix }}/static/eonasdan-bootstrap-datetimepicker/bootstrap-datetimepicker.min.js"></script>
+    <script src="{{ pathPrefix }}/static/vendor/bootstrap3-typeahead/bootstrap3-typeahead.js"></script>
 
-    <script src="{{ pathPrefix }}/static/vendor/js/handlebars.js"></script>
+    <script src="{{ pathPrefix }}/static/handlebars/handlebars.js"></script>
     <script src="{{ pathPrefix }}/static/vendor/js/jquery.selection.js"></script>
-    <script src="{{ pathPrefix }}/static/vendor/js/jquery.hotkeys.js"></script>
+    <script src="{{ pathPrefix }}/static/jquery-hotkeys/jquery.hotkeys.js"></script>
 
     <script src="{{ pathPrefix }}/static/js/graph.js?v=1"></script>
 
--- a/web/ui/templates/_base.html
+++ b/web/ui/templates/_base.html
@@ -3,10 +3,10 @@
   <head>
     <meta http-equiv="Content-Type" content="text/html; charset=utf-8">
     <title>Prometheus Time Series Collection and Processing Server</title>
-    <script src="{{ pathPrefix }}/static/vendor/js/jquery.min.js"></script>
-    <script src="{{ pathPrefix }}/static/vendor/bootstrap-3.3.1/js/bootstrap.min.js"></script>
+    <script src="{{ pathPrefix }}/static/jquery/jquery.min.js"></script>
+    <script src="{{ pathPrefix }}/static/bootstrap/js/bootstrap.min.js"></script>
 
-    <link type="text/css" rel="stylesheet" href="{{ pathPrefix }}/static/vendor/bootstrap-3.3.1/css/bootstrap.min.css">
+    <link type="text/css" rel="stylesheet" href="{{ pathPrefix }}/static/bootstrap/css/bootstrap.min.css">
     <link type="text/css" rel="stylesheet" href="{{ pathPrefix }}/static/css/prometheus.css">
 
     <script>
--- a/cmd/prometheus/config.go
+++ b/cmd/prometheus/config.go
@@ -92,6 +92,11 @@
 		"Path under which to expose metrics.",
 	)
 	cfg.fs.StringVar(
+		&cfg.web.LocalAssets, "web.local-assets",
+		"/usr/share/prometheus/web/",
+		"Path to static assets/templates directory.",
+	)
+	cfg.fs.StringVar(
 		&cfg.web.UserAssetsPath, "web.user-assets", "",
 		"Path to static asset directory, available at /user.",
 	)
--- a/web/ui/static/js/graph.js
+++ b/web/ui/static/js/graph.js
@@ -110,13 +110,18 @@
 
   self.endDate = graphWrapper.find("input[name=end_input]");
   self.endDate.datetimepicker({
-    language: 'en',
-    pickSeconds: false,
+    locale: 'en',
+    format: 'YYYY-MM-DD HH:mm',
+    toolbarPlacement: 'bottom',
+    sideBySide: true,
+    showTodayButton: true,
+    showClear: true,
+    showClose: true,
   });
   if (self.options.end_input) {
-    self.endDate.data('datetimepicker').setValue(self.options.end_input);
+    self.endDate.data('DateTimePicker').date(self.options.end_input);
   }
-  self.endDate.change(function() { self.submitQuery(); });
+  self.endDate.on("dp.change", function() { self.submitQuery(); });
   self.refreshInterval.change(function() { self.updateRefresh(); });
 
   self.isStacked = function() {
@@ -277,9 +282,9 @@
 Prometheus.Graph.prototype.getEndDate = function() {
   var self = this;
   if (!self.endDate || !self.endDate.val()) {
-    return new Date();
+    return moment();
   }
-  return self.endDate.data('datetimepicker').getDate().getTime();
+  return self.endDate.data('DateTimePicker').date();
 };
 
 Prometheus.Graph.prototype.getOrSetEndDate = function() {
@@ -291,18 +296,22 @@
 
 Prometheus.Graph.prototype.setEndDate = function(date) {
   var self = this;
-  self.endDate.data('datetimepicker').setValue(date);
+  self.endDate.data('DateTimePicker').date(date);
 };
 
 Prometheus.Graph.prototype.increaseEnd = function() {
   var self = this;
-  self.setEndDate(new Date(self.getOrSetEndDate() + self.parseDuration(self.rangeInput.val()) * 1000/2 )); // increase by 1/2 range & convert ms in s
+  var newDate = moment(self.getOrSetEndDate());
+  newDate.add(self.parseDuration(self.rangeInput.val()) / 2, 'seconds');
+  self.setEndDate(newDate);
   self.submitQuery();
 };
 
 Prometheus.Graph.prototype.decreaseEnd = function() {
   var self = this;
-  self.setEndDate(new Date(self.getOrSetEndDate() - self.parseDuration(self.rangeInput.val()) * 1000/2 ));
+  var newDate = moment(self.getOrSetEndDate());
+  newDate.subtract(self.parseDuration(self.rangeInput.val()) / 2, 'seconds');
+  self.setEndDate(newDate);
   self.submitQuery();
 };
 
